### DISCUSSION TITLE
Spatie\MediaLibrary\MediaCollections\Exceptions\UnreachableUrl: Url

### DISCUSSION THREAD

**Discussion by aliraza371** [spatie/laravel-medialibrary#2313] [General]
Hi, 

I'm running a job that fetches records from the third-party system and I'm using this library to download each record image as well with addMediaFromUrl(). It's working fine but when the remote URL does not have an image its throws UnreachableUrl: Url exception and fails my Queue Job. 

Any can explain how could I handle a URL that does not have media without a failing process?

**Comment by OccultScientist** [comment]
Hey, I'm having the same issue.
Did you ever find a solution?
